### ISSUE TITLE
Implement metrics

### ISSUE THREAD

**Issue by kevinsuedmersen** [kevinsuedmersen/library-neuralnetworksfromscratch#16]
- Implement the following metrics
    - [x] Accuracy
    - [x] Precision
    - [x] Recall
    - [x] Categorical Cross Entropy: Figure out why it's not working yet [link](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html)
- Apply metrics
    - [x] Evaluate metrics every x training batches
    - [x] Implement `val_step`
- Test Suite fixes
    - [x] Fix rounding errors (see `test_model.py` line 249
